src/scipp/core/data_group.py

```diff
@@ -191,49 +195,48 @@ def __add__(self, other):
     def __mul__(self, other):
         return _data_group_binary(operator.mul, self, other)
 
-    def _call_method(self, func: Callable) -> DataGroup:
-        """Call method on all values and return new DataGroup containing the results."""
-        return DataGroup({key: func(value) for key, value in self.items()})
-
     @property
     def bins(self):
         # TODO Returning a regular DataGroup here may be wrong, since the `bins`
         # property provides a different set of attrs and methods.
-        return self._call_method(operator.attrgetter('bins'))
+        return self.apply(operator.attrgetter('bins'))
+
+    def apply(self, func: Callable, *args, **kwargs) -> DataGroup:
```

Comment:
Are the extra arguments necessary? We could require callers to use `partial` or similar.

Reply:
Are they a problem? I used a signature similar to Pandas.

Reply:
Not a problem. But it would simplify the interface. Leave it, I just wanted to know your motivation.